File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/loader/processors/gbd/DummyProgress.java
package ru.kmorozov.library.data.loader.processors.gbd;

import ru.kmorozov.gbd.logger.progress.IProgress;

public class DummyProgress implements IProgress {

    @Override
    public int inc() {
        return 0;
    }

    @Override
    public int incrementAndProgress() {
        return 0;
    }

    @Override
    public void finish() {

    }

    @Override
    public IProgress getSubProgress(final int maxValue) {
        return new DummyProgress();
    }

    @Override
    public void resetMaxValue(final int maxValue) {

    }
}

File: ru.kmorozov.library.data.server/src/main/java/ru/kmorozov/library/data/loader/processors/gbd/DummyProgress.kt
package ru.kmorozov.library.data.loader.processors.gbd

import ru.kmorozov.gbd.logger.progress.IProgress

class DummyProgress : IProgress {

    override fun inc(): Int {
        return 0
    }

    override fun incrementAndProgress(): Int {
        return 0
    }

    override fun finish() {

    }

    override fun getSubProgress(maxValue: Int): IProgress {
        return DummyProgress()
    }

    override fun resetMaxValue(maxValue: Int) {

    }
}
